shared/actions/git.js

```diff
@@ -2,211 +2,108 @@
-  }
-}
-
-// reset errors and set loading, make a call and either go back to the root or show an error
-function* _createDeleteHelper(theCall: any): Generator<any, void, any> {
-  yield Saga.put.resolve(GitGen.createSetError({error: null}))
-  yield Saga.put.resolve(WaitingGen.createIncrementWaiting({key: Constants.loadingWaitingKey}))
-  try {
-    yield theCall
-    yield Saga.put(navigateTo(isMobile ? [Tabs.settingsTab, SettingsConstants.gitTab] : [Tabs.gitTab], []))
-    yield Saga.put(GitGen.createLoadGit())
-  } catch (err) {
-    yield Saga.put(GitGen.createSetError({error: err}))
-  } finally {
-    // just in case
-    yield Saga.put.resolve(WaitingGen.createDecrementWaiting({key: Constants.loadingWaitingKey}))
-  }
-}
-
-const _createPersonalRepo = (action: GitGen.CreatePersonalRepoPayload) =>
-  Saga.call(
-    _createDeleteHelper,
-    Saga.call(RPCTypes.gitCreatePersonalRepoRpcPromise, {
-      repoName: action.payload.name,
-    })
-  )
-
-const _createTeamRepo = (action: GitGen.CreateTeamRepoPayload) =>
-  Saga.call(
-    _createDeleteHelper,
-    Saga.call(RPCTypes.gitCreateTeamRepoRpcPromise, {
-      notifyTeam: action.payload.notifyTeam,
-      repoName: action.payload.name,
-      teamName: {
-        parts: action.payload.teamname.split('.'),
+const load = (state: TypedState) =>
+  state.config.loggedIn &&
+  RPCTypes.gitGetAllGitMetadataRpcPromise(undefined, Constants.loadingWaitingKey)
+    .then((results: ?Array<RPCTypes.GitRepoResult>) =>
+      GitGen.createLoaded(Constants.parseRepos(results || []))
+    )
+    .catch(() => {})
+
+const loadGitRepo = (state: TypedState, action: GitGen.LoadGitRepoPayload) =>
+  state.config.loggedIn &&
+  RPCTypes.gitGetGitMetadataRpcPromise(
+    {
+      folder: {
+        created: false,
+        folderType: action.payload.teamname
+          ? RPCTypes.favoriteFolderType.team
+          : RPCTypes.favoriteFolderType.private,
+        name: action.payload.teamname || action.payload.username || '',
+        notificationsOn: false,
+        private: true,
       },
-    })
-  )
-
-const _deletePersonalRepo = (action: GitGen.DeletePersonalRepoPayload) =>
-  Saga.call(
-    _createDeleteHelper,
-    Saga.call(RPCTypes.gitDeletePersonalRepoRpcPromise, {
-      repoName: action.payload.name,
-    })
+    },
+    Constants.loadingWaitingKey
   )
-
-const _deleteTeamRepo = (action: GitGen.DeleteTeamRepoPayload) =>
-  Saga.call(
-    _createDeleteHelper,
-    Saga.call(RPCTypes.gitDeleteTeamRepoRpcPromise, {
-      notifyTeam: action.payload.notifyTeam,
+    .then((results: ?Array<RPCTypes.GitRepoResult>) =>
+      GitGen.createLoaded(Constants.parseRepos(results || []))
+    )
+    .catch(() => {})
+
+const surfaceGlobalErrors = (_, {payload: {errors}}: GitGen.LoadedPayload) =>
+  Saga.all(errors.map(globalError => Saga.put(ConfigGen.createGlobalError({globalError}))))
+
+// Do we have to call loadgit?
+const createPersonalRepo = (_, action: GitGen.CreatePersonalRepoPayload) =>
```

Comment:
The old code called `loadGit` after it finished the rpc message. It's isn't necessary because the component calls `loadGit` on mount.

If we do want this behavior anyways we could have this dispatch an action like `reposModified` that loadsGit and Navs